pkg/apis/query/v0alpha1/query.go

```diff
@@ -34,6 +34,9 @@ func GetResponseCode(rsp *backend.QueryDataResponse) int {
 	}
 	for _, res := range rsp.Responses {
 		if res.Error != nil {
+			if res.Status > 0 {
+				return int(res.Status)
+			}
```

Comment:
i see what we're trying to do here, but  there is the problem of, if there are multiple errors with different error-codes, what should happen? (i think you also mentioned this).

if possible, can we deal with this change in a separate PR, so that we can discuss it separately? or do other things in the PR depend on this?

Reply:
yeah after our chat this morning I don't think this is necessary. If we always return a k8s error for 404s then this is not needed, removing